test/unit/spec/EnrollCustomFactor_spec.js

```diff
@@ -0,0 +1,132 @@
+/* eslint max-params:[2, 14] */
+define([
+  'okta',
+  '@okta/okta-auth-js/jquery',
+  'util/Util',
+  'shared/util/Util',
+  'helpers/mocks/Util',
+  'helpers/dom/EnrollCustomFactorForm',
+  'helpers/dom/Beacon',
+  'helpers/util/Expect',
+  'sandbox',
+  'LoginRouter',
+  'helpers/xhr/MFA_ENROLL_allFactors',
+  'helpers/xhr/MFA_ENROLL_ACTIVATE_CustomFactor',
+  'helpers/xhr/customFactor_error',
+  'helpers/xhr/SUCCESS'
+],
+function (Okta,
+          OktaAuth,
+          LoginUtil,
+          SharedUtil,
+          Util,
+          Form,
+          Beacon,
+          Expect,
+          $sandbox,
+          Router,
+          responseMfaEnrollAll,
+          responseMfaEnrollActivateCustomFactor,
+          resCustomFactorError,
+          responseSuccess) {
+
+  var itp = Expect.itp;
+  var tick = Expect.tick;
+
+  Expect.describe('EnrollCustomFactor', function () {
+
+    function setup() {
+      var setNextResponse = Util.mockAjax([responseMfaEnrollAll, responseMfaEnrollActivateCustomFactor]);
+      var baseUrl = 'https://foo.com';
+      var authClient = new OktaAuth({url: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR});
+      var successSpy = jasmine.createSpy('success');
+      var router = new Router({
+        el: $sandbox,
+        baseUrl: baseUrl,
+        authClient: authClient,
+        globalSuccessFn: successSpy
+      });
+      Util.registerRouter(router);
+      Util.mockRouterNavigate(router);
+      return tick()
+      .then(function () {
+        router.refreshAuthState('dummy-token');
+        return Expect.waitForEnrollChoices();
+      })
+      .then(function () {
+        router.enrollCustomFactor();
+        return Expect.waitForEnrollCustomFactor({
+          router: router,
+          beacon: new Beacon($sandbox),
+          form: new Form($sandbox),
+          ac: authClient,
+          setNextResponse: setNextResponse,
+          successSpy: successSpy
+        });
+      });
+    }
+
+    Expect.describe('Header & Footer', function () {
+      itp('displays the correct factorBeacon', function () {
+        return setup().then(function (test) {
+          expect(test.beacon.isFactorBeacon()).toBe(true);
+          expect(test.beacon.hasClass('mfa-custom-factor')).toBe(true);
+        });
+      });
+      itp('has a "back" link in the footer', function () {
+        return setup().then(function (test) {
+          Expect.isVisible(test.form.backLink());
+        });
+      });
+    });
+
+    Expect.describe('Enroll factor', function () {
+
+      itp('displays correct title', function () {
+        return setup().then(function (test) {
+          test.setNextResponse(responseSuccess);
+          expect(test.form.titleText())
+            .toBe('GENERIC_SAML');
```

Comment:
can we change the `vendorName` in the  test xhr files to be something different than the provider? So we can test we are displaying the correct name. Maybe something like `Third Party Factor`?

Reply:
ok 👍